### PR TITLE
Patch Rabit issue causing slowdown of XGBoost on OS X

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

**Review by Pscheidl — APPROVED** [review]

**Comment by Pscheidl** [comment]
@michalkurka Approved, definitely an improvement. Sleep deprivation is better to be avoided. :+1: :rofl:  Are we waiting for dmlc/xgboost to provide the ultimate fix ?

File: ci/Makefile.jenkins [hunk 1]
```diff
@@ -140,13 +140,17 @@ endif
 mrproper:
 	make clean_all
 	git clean -f -d -x
+	$(MAKE) -f $(THIS_FILE) apply_patches
```
Comment (michal-raska) [inline ci/Makefile.jenkins:143]: Minor comment, I'm not sure this should be part of mrproper. I was always looking at mrproper as "clean everything" target, but I might be wrong.

**Review by michal-raska — APPROVED** [review]

File: ci/Makefile.jenkins (hunk 1 above)
Comment (michalkurka) [reply ci/Makefile.jenkins:143]: I think you are right... my interpretation of `mrproper` is to restore to a clean state. If we had a fork of the Rabit repository, the clean state would include a commit that has the patch. This is just a workaround that we don't have the fork and I cannot just make a commit.

IMHO this is not common but maybe justifiable? :)

I will leave it for now and we can figure out a better way - ideally, make a PR to the Rabit repo.

**Comment by michalkurka** [comment]
@Pscheidl yes, this should be fixed in XGBoost but this will solve this issue for us. The only penalty we incur is when a multi-node model training is finished. This fixes issues with scoring regardless of the number of nodes because scoring in H2O is always single-node (per-chunk).

Makes me think we are good now.

**Pull request merged**